activesupport/lib/active_support/duration/iso8601_serializer.rb

```diff
@@ -40,6 +42,14 @@ def normalize
           parts = @duration.parts.each_with_object(Hash.new(0)) do |(k, v), p|
             p[k] += v  unless v.zero?
           end
+
+          # Convert weeks to days and remove weeks if mixed with date parts
+          if week_mixed_with_date?(parts)
+            days_in_week = SECONDS_PER_WEEK / SECONDS_PER_DAY
+            parts[:days] = parts[:days].to_i + (parts[:weeks] * days_in_week)
```

Comment:
Why `parts[:days].to_i`?

Reply:
Thanks for pointing it out, have updated